lib/optimizely.rb

```diff
@@ -267,6 +267,11 @@ def is_feature_enabled(feature_flag_key, user_id, attributes = nil)
         variation = decision['variation']
         # Send event if Decision came from an experiment.
         if decision.source == Optimizely::DecisionService::DECISION_SOURCE_EXPERIMENT
+          unless variation[Optimizely::Helpers::Constants::FEATURE_ENABLED]
+            @logger.log(Logger::INFO,
+                        "Feature '#{feature_flag_key}' is not enabled for user '#{user_id}'.")
+            return false
+          end
```

Comment:
Please refactor this code and make it easier. 
Move: ' # Send event if Decision came from an experiment.' to place where impression is being sent